### PR TITLE
(maint) Update gems that ship with bolt

### PR DESCRIPTION
**Pull request by donoghuc**
Update gems that ship with bolt system packages. The most important update in this commit is the the latest puppet_resource-api which contains important bug fixes for supporting device modules.

### CODE REVIEW

**Comment by donoghuc** [comment]
cas donoghue [11:50 AM]
!mirrorsource https://rubygems.org/downloads/puppet-resource_api-1.8.4.gem

Jira BotAPP [11:50 AM]
Let's get you connected to Jira.  It's a simple process to get unlocked :old_key:

Connect to Jira

kerminatorAPP [11:50 AM]
@casdonoghue: Success! https://artifactory.delivery.puppetlabs.net/artifactory/generic__buildsources/buildsources/puppet-resource_api-1.8.4.gem md5: bc43e38b0a90638c959994f0223744f6

**Comment by donoghuc** [comment]
cas donoghue [8:13 AM]
!mirrorsource https://rubygems.org/downloads/ruby_smb-1.1.0.gem

kerminatorAPP [8:13 AM]
@casdonoghue: Success! https://artifactory.delivery.puppetlabs.net/artifactory/generic__buildsources/buildsources/ruby_smb-1.1.0.gem md5: f926be4cb045b5984d02dc10e2b240ca
cas donoghue  [23 minutes ago]
!mirrorsource https://rubygems.org/downloads/puppet-6.6.0.gem

kerminatorAPP   [23 minutes ago]
@casdonoghue: Success! https://artifactory.delivery.puppetlabs.net/artifactory/generic__buildsources/buildsources/puppet-6.6.0.gem md5: bc80b8ff8b92758d24fdc50f82121ff6

cas donoghue  [22 minutes ago]
!mirrorsource https://rubygems.org/downloads/public_suffix-3.1.1.gem

kerminatorAPP   [22 minutes ago]
@casdonoghue: Success! https://artifactory.delivery.puppetlabs.net/artifactory/generic__buildsources/buildsources/public_suffix-3.1.1.gem md5: df5c3a90582565bb91b035c771a58075

cas donoghue  [17 minutes ago]
!mirrorsource https://rubygems.org/downloads/gettext-3.2.9.gem

kerminatorAPP   [17 minutes ago]
@casdonoghue: Uh oh... An error has occurred: gettext-3.2.9.gem already exists on Artifactory. md5: 09a755cd03ab617835e20a2e910581f4

cas donoghue  [7 minutes ago]
!mirrorsource https://rubygems.org/downloads/deep_merge-1.2.1.gem

kerminatorAPP   [7 minutes ago]
@casdonoghue: Success! https://artifactory.delivery.puppetlabs.net/artifactory/generic__buildsources/buildsources/deep_merge-1.2.1.gem md5: 8d8396705375ac646454b1d64ad1239a

File: configs/components/rubygem-gettext.rb [hunk 1]
```diff
@@ -1,6 +1,16 @@
 component "rubygem-gettext" do |pkg, settings, platform|
-  pkg.version "3.2.2"
-  pkg.md5sum "4cbb125f8d8206e9a8f3a90f6488e4da"
+  # Projects may define a :rubygem_gettext_version setting, or we use 3.2.2 by default:
+  version = settings[:rubygem_gettext_version] || '0.6.1'
```
Comment (lucywyman) [inline configs/components/rubygem-gettext.rb]: Did this intend to be `3.2.2` instead of `0.6.1`?

Comment (donoghuc) [reply configs/components/rubygem-gettext.rb]: ugh. I fixed that locally but forgot to push it. That is what I get putting up PRs before I have coffee in the morning.

**Comment by donoghuc** [comment]
Bolt runtime tarballs: http://builds.delivery.puppetlabs.net/puppet-runtime/4ce4c57ba5ac23d68e08cb876c3fea8062673544/artifacts/?C=M&O=D
Bolt packages: http://builds.delivery.puppetlabs.net/puppet-bolt/b42ad64f668d0787a1af1b52616e5c4a7b62e145/artifacts/?C=M&O=D
Packaging acceptance tests: https://jenkins-master-prod-1.delivery.puppetlabs.net/view/__experimental%20automatic/job/experimental_auto_bolt-vanagon_vanagon-integration-suite_no-PEZ-daily-master/

**Pull request merged**